internal/pubsub/gcppubsub/gcppubsub.go

```diff
@@ -57,19 +89,52 @@ func (t *topic) SendBatch(ctx context.Context, dms []*driver.Message) error {
 }
 
 type subscription struct {
-	client *rawgcppubsub.SubscriberClient
+	client *raw.SubscriberClient
 	path   string
 }
 
-// ReceiveBatch returns a batch of messages that have queued up for the
-// subscription on the server. If no messages are available yet, it blocks
-// until there is at least one, or the context is done.
+// SubscriptionOptions will contain configuration for subscriptions.
+type SubscriptionOptions struct{}
+
+// OpenSubscription opens the subscription on GCP PubSub for the given
+// projectID and subscriptionName. If the subscription does not exist then an
+// error is returned.
+func OpenSubscription(ctx context.Context, client *raw.SubscriberClient, proj gcp.ProjectID, subscriptionName string, opts *SubscriptionOptions) (*pubsub.Subscription, error) {
+	ds, err := openSubscription(ctx, client, proj, subscriptionName)
+	if err != nil {
+		return nil, fmt.Errorf("opening subscription: %v", err)
+	}
+	s := pubsub.NewSubscription(ctx, ds)
+	return s, nil
+}
+
+func openSubscription(ctx context.Context, client *raw.SubscriberClient, projectID gcp.ProjectID, subscriptionName string) (driver.Subscription, error) {
+	gcli, err := raw2.NewClient(ctx, string(projectID))
+	if err != nil {
+		return nil, fmt.Errorf("making GCP client to check subscription existence: %v", err)
+	}
+	gs := gcli.Subscription(subscriptionName)
+	ok, err := gs.Exists(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("checking for existence of subscription: %v", err)
+	}
+	if !ok {
+		return nil, fmt.Errorf("subscription named %q does not exist", subscriptionName)
+	}
+	path := fmt.Sprintf("projects/%s/subscriptions/%s", projectID, subscriptionName)
+	return &subscription{client, path}, nil
+}
+
+// ReceiveBatch implements driver.Subscription.ReceiveBatch.
 func (s *subscription) ReceiveBatch(ctx context.Context) ([]*driver.Message, error) {
-	req := &pubsubpb.PullRequest{
+	req := &pb.PullRequest{
 		Subscription:      s.path,
 		ReturnImmediately: false,
 	}
 	resp, err := s.client.Pull(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("making RPC to receive next batch of messages: %v", err)
```

Comment:
If we wrap errors like this, it will be impossible to tell if they're retryable.

Reply:
Done.